contracts/xrpl-multisig-prover/src/types.rs

```diff
@@ -0,0 +1,132 @@
+use axelar_wasm_std::nonempty;
+use connection_router::state::CrossChainId;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{from_binary, HexBinary, StdResult, Uint256};
+use cw_storage_plus::{Key, KeyDeserialize, PrimaryKey};
+use multisig::key::Signature;
+use xrpl_voting_verifier::execute::MessageStatus;
+
+use crate::{xrpl_multisig::XRPLUnsignedTx, error::ContractError};
+
+#[cw_serde]
+pub enum TransactionStatus {
+    Pending,
+    Succeeded,
+    FailedOnChain,
+    FailedOffChain,
+}
+
+// TODO: import from verifier
+pub fn parse_message_id(
+    message_id: &nonempty::String,
+) -> Result<(nonempty::String, u64), ContractError> {
+    // expected format: <tx_id>:<index>
+    let components = message_id.split(":").collect::<Vec<_>>();
```

Comment:
This is not a standard format for message ids. This is used by evm chains, but other chains could use different formats. What types of messages is this called on specifically?